tests/unit/s2n_ktls_test.c

```diff
@@ -15,40 +15,125 @@
 
 #include "tls/s2n_ktls.h"
 
+#include "error/s2n_errno.h"
 #include "s2n_test.h"
 #include "testlib/s2n_testlib.h"
+#include "utils/s2n_random.h"
+#include "utils/s2n_socket.h"
 
-S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
+#define S2N_TEST_SEND_FD 66
+#define S2N_TEST_RECV_FD 55
 
-/* set kTLS supported cipher */
-struct s2n_cipher ktls_temp_supported_cipher = {
-    .ktls_supported = true,
-};
-struct s2n_record_algorithm ktls_temp_supported_record_alg = {
-    .cipher = &ktls_temp_supported_cipher,
-};
-struct s2n_cipher_suite ktls_temp_supported_cipher_suite = {
-    .record_alg = &ktls_temp_supported_record_alg,
-};
+#if defined(S2N_KTLS_SUPPORTED)
+/* Its difficult to test this method via s2n_connection_ktls_enable_send/recv because
+ * the key_material is populated via prf, which tends to be non-deterministic */
```

Comment:
Nit
```suggestion
/* It's difficult to test this method via s2n_connection_ktls_enable_send/recv because
 * the key_material is populated via prf, which tends to be non-deterministic */
```
Might also want to say `which by definition produces "pseudo-random" output` instead, because I think that captures your issue better.